scripts/main/photo.js

```diff
@@ -1021,6 +1021,49 @@ photo.getArchive = function (photoIDs, kind = null) {
 	location.href = "api/Photo::getArchive" + lychee.html`?photoIDs=${photoIDs.join()}&kind=${kind}`;
 };
 
+/**
+ * Shows a dialog to share the view URL via a QR code.
+ *
+ * @param {string} photoID
+ * @returns {void}
+ */
+photo.qrCode = function (photoID) {
+	let myPhoto;
+
+	if (photo.json && photo.json.id === photoID) {
+		myPhoto = photo.json;
+	} else {
+		myPhoto = album.getByID(photoID);
+	}
```

Comment:
Using the tertiary operator allows to use `const` for the variable.

```suggestion
	/** @type {?Photo} */
	const myPhoto = (photo.json && photo.json.id === photoID) ? photo.json : album.getByID(photoID);
```

Also note, that there is a small chance that `album.getByID` returns `null`. You must check for that.

Reply:
https://github.com/LycheeOrg/Lychee-front/pull/287/commits/66b5448ce0939fa679ef6a2b1c52b5140227f762